src/jit/decomposelongs.cpp

```diff
@@ -796,11 +799,9 @@ GenTree* DecomposeLongs::DecomposeInd(LIR::Use& use)
         addrBase->TypeGet(), addrBase->AsLclVarCommon()->GetLclNum(), BAD_IL_OFFSET);
     GenTreePtr addrHigh = new(m_compiler, GT_LEA) GenTreeAddrMode(TYP_REF, addrBaseHigh, nullptr, 0, genTypeSize(TYP_INT));
     GenTreePtr indHigh = new (m_compiler, GT_IND) GenTreeIndir(GT_IND, TYP_INT, addrHigh, nullptr);
-    
+
     // Insert the nodes into the block
-    m_blockRange.InsertAfter(addrBaseHigh, indLow);
-    m_blockRange.InsertAfter(addrHigh, addrBaseHigh);
-    m_blockRange.InsertAfter(indHigh, addrHigh);
+    m_blockRange.InsertAfter(LIR::SeqTree(m_compiler, indHigh), indLow);
```

Comment:
There's a third case: 3. a new local var ref is created to an operand.

In the StoreInd case, the operand is removed from the range before the new tree is sequenced, then re-inserted by the InsertAfter code ("code motion" of the local/leaf), so SeqTree would be safe. In the DecomposeInd case, there is also no sharing, as the "addrBaseHigh" creates a new GT_LCL_VAR reference.

I'm ok using gtPrepareCosts + a sequence of node InsertAfter, but it's more lines of code and less easy to read/reason about, IMO.